builder/bundleset.go

```diff
@@ -345,7 +364,20 @@ func parseBundle(contents []byte) (*bundle, error) {
 		return nil, scanner.Err()
 	}
 
+	// if a bundle was added to both includes and optional
+	// remove it from the optional
+	for i, optionalBundle := range optional {
+		for _, includedBundle := range includes {
+			if optionalBundle == includedBundle {
+				optional[i] = optional[len(optional)-1]
+				optional[len(optional)-1] = ""
+				optional = optional[:len(optional)-1]
```

Comment:
I don't think that this is safe.

* `optional[i] = optional[len(optional)-1]`
  * The last bundle in the slice is accepted without confirming that it is not in the includedBundle slice.
* `optional[len(optional)-1] = ""`
  * This line can be removed. In the following line, a new slice is created that does not include the underlying array element at index len(optional)-1.

Also, we need to be careful about modifying the variable that we range over. I think `range` creates a copy of the values in the slice and iterates over them, so it will not notice changes to that occur while iterating.

Reply:
It is probably better to set the values that needs to be removed to `""` and them trim them out of the slice afterwards. This will also preserve the order.